rmw_connextdds_common/src/ndds/dds_api_ndds.cpp

```diff
@@ -1275,3 +1312,120 @@ rmw_connextdds_enable_security(
 
   return RMW_RET_OK;
 }
+
+rmw_ret_t
+rmw_connextdds_set_cft_filter_expression(
+  DDS_TopicDescription * const topic_desc,
+  const char * filter_expression,
+  const rcutils_string_array_t * expression_parameters)
+{
+  DDS_ContentFilteredTopic * const cft_topic =
+    DDS_ContentFilteredTopic_narrow(topic_desc);
+
+  struct DDS_StringSeq cft_parameters = DDS_SEQUENCE_INITIALIZER;
+  auto scope_exit_cft_parameters = rcpputils::make_scope_exit(
+    [&cft_parameters]() {
+      if (!DDS_StringSeq_finalize(&cft_parameters)) {
+        RMW_CONNEXT_LOG_ERROR_SET("failed to finalize cft parameters sequence")
+      }
+    });
+  if (expression_parameters) {
+    if (RMW_RET_OK !=
+      rmw_connextdds_initialize_cft_parameters(&cft_parameters, expression_parameters))
+    {
+      RMW_CONNEXT_LOG_ERROR_SET("failed to rmw_connextdds_initialize_cft_parameters")
+      return RMW_RET_ERROR;
+    }
+  }
+
+  DDS_ReturnCode_t ret =
+    DDS_ContentFilteredTopic_set_expression(cft_topic, filter_expression, &cft_parameters);
+  if (DDS_RETCODE_OK != ret) {
+    RMW_CONNEXT_LOG_ERROR_SET("failed to set content-filtered topic")
+    return RMW_RET_ERROR;
+  }
+  return RMW_RET_OK;
+}
+
+rmw_ret_t
+rmw_connextdds_get_cft_filter_expression(
+  DDS_TopicDescription * const topic_desc,
+  char ** const expr_out,
+  rcutils_string_array_t * cft_params_out)
+{
+  DDS_ContentFilteredTopic * const cft_topic =
+    DDS_ContentFilteredTopic_narrow(topic_desc);
+
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+
+  // get filter_expression
+  const char * expression = DDS_ContentFilteredTopic_get_filter_expression(cft_topic);
+  if (!expression) {
+    RMW_SET_ERROR_MSG("failed to get filter expression");
+    return RMW_RET_ERROR;
+  }
+
+  *expr_out = rcutils_strdup(expression, allocator);
+  if (NULL == *expr_out) {
+    RMW_SET_ERROR_MSG("failed to duplicate string");
```

Comment:
Sorry, but I hadn't noticed you used these different macros here.

Please replace `RMW_SET_ERROR_MSG(...)` with `RMW_CONNEXT_LOG_ERROR_SET(...)`, and `RMW_SET_ERROR_MSG_WITH_FORMAT_STRING(...)` with `RMW_CONNEXT_LOG_ERROR_A_SET(...)`.

You should also remove the `;` from the log lines after switching macros.

Reply:
Don't do that, I think I should say 'sorry'. Thanks for your review, it's very nice of you.

Reply:
🙏🏼 It's my pleasure to help you, I'm very glad to have others interested in this software!